### PR TITLE
Place toolbar inside blockly container div

### PR DESCRIPTION
**Pull request by sravan-s**
**blocklyToolboxDiv** was previously injected into body, causing some issues with certain page layouts.
In this commit, this specific div has been set inside the container where blckly is injected and by setting proper styling, positioning is maintained.

### CODE REVIEW

**Comment by jano42** [comment]
This is also done in #241

**Comment by sravan-s** [comment]
Ok...  I checked #241, why aren't they merging it? I made this fix because bockly was giving us some trouble in our polymer project. The toolbar remains visible, even after we changed pages.

**Comment by jano42** [comment]
I don't know why they haven't merged yet.
It make some month that I'm maintain it with last master commits...

I expect this pr to be merged asap...

**Comment by sravan-s** [comment]
@jano42 Can I do anything to get Blockly maintainer's attention to your PR? Any idea who is maintaining Blockly?

**Comment by rachel-fenichel** [comment]
@sravan-s This looks good in the playground and block factory, but breaks in the multi-playground.  Please take a look at that test case.

**Comment by sravan-s** [comment]
@rachel-fenichel Okie, will look into that. Does #241 got the same issue?

**Comment by tmickel** [comment]
@picklesrus I'm worried this might conflict with some of your upcoming changes?

**Comment by rachel-fenichel** [comment]
@sravan-s I'm not sure but I suspect it does.  It looks very similar to this one.

**Comment by sravan-s** [comment]
@rachel-fenichel I checked that, actually, it's related to some css in **multi_playground.html**.

```
#octaweb td  div {
  height: 480px;
  width: 100%;
}
```

Sets every **<div>** inside the **<td>** tag with 100% width(Now toolbars are inside blockly, so they also takes 100% width and 480px height). I set  the rue as **#octaweb td  > div** , it fixes the issue. Can you verify?

@tmickel Where can I find more info on the upcoming changes?

**Comment by rachel-fenichel** [comment]
@sravan-s: verified.  Looks good to me, but I'm still going to pass it on to get another opinion before merging.

**Comment by sravan-s** [comment]
@rachel-fenichel yes, understood.

**Comment by picklesrus** [comment]
@sravan-s The changes @tmickel was referring to are an attempt to improve workspace performance by reorganizing the svgs in blockly. This conflicts in the sense that I change the same lines as you did in a slightly different way, but I should be able to adjust.

**Comment by sravan-s** [comment]
So... Is this good to go?

**Comment by RoboErikG** [comment]
Hey sravan-s, sorry for the delay. We need Neil to take a quick look first but he's out this week. He said he might check in a bit during the week so we might get to it sooner.

File: core/toolbox.js [hunk 1]
```diff
@@ -233,25 +233,24 @@ Blockly.Toolbox.prototype.position = function() {
   var svgPosition = goog.style.getPageOffset(svg);
   var svgSize = Blockly.svgSize(svg);
   if (this.horizontalLayout_) {
-    treeDiv.style.left = svgPosition.x + 'px';
+    treeDiv.style.left = '0px';
```
Comment (NeilFraser) [inline core/toolbox.js]: These should just be 0, in accordance with the Google style guide:
https://google.github.io/styleguide/htmlcssguide.xml?showone=0_and_Units#0_and_Units

Comment (sravan-s) [inline core/toolbox.js]: Fixed

**Comment by RoboErikG** [comment]
Just a quick update. We're going to pull this change into Blockly Games to test and make sure everything still works as expected. If everything works we'll merge and then circle back to #241 to see if there's additional features we'd like to support. ETA is probably next week.

**Comment by NeilFraser** [comment]
A quick test of this PR in Blockly Games shows that Blockly renders like this:
toolbox2
For reference, I'd expect Blockly to render like this:
toolbox1

Another bug can be easily seen in the Block Factory.  First load looks fine.  Now start resizing the window and watch Blockly grow progressively bigger:
bf

**Comment by sravan-s** [comment]
@NeilFraser I'll check those bugs. But I am not being able to properly understand the second bug(_Blockly grow progressively bigger_) you pointed out. Can you please make it a bit more clearer?

**Comment by NeilFraser** [comment]
@sravan-s Here's a video showing the bug in the Block Factory as the window is resized.  It also shows that the Code demo is also broken.
https://youtu.be/i9gdDvUqbOU

Firefox on OS X (Chrome on OS X is the same).

**Comment by jano42** [comment]
This resize bug is laso fix in #241 
This is just a simple css trick

`````css
  '.blocklySvg {',
    'background-color: #fff;',
    'outline: none;',
    'overflow: hidden;',  /* IE overflows by default. */
	'display: block;',  /* fix svg area growup bug */
  '}',
````````

**Comment by NeilFraser** [comment]
#241 seems to do what this PR does, and appears to be bug-free.  I suggest accepting #241 and closing #492.  Is there anything in #492 than we should move over to #241?

**Comment by sravan-s** [comment]
@NeilFraser It's cool. I think #241 is a super-set of this PR. I'll close this.

**Pull request closed without merging**